internal/core/metadata/application/notify.go

```diff
@@ -78,6 +78,8 @@ func validateDeviceCallback(ctx context.Context, dic *di.Container, device dtos.
 		// allow this case for the backward-compatability in v2
 		if err.Code() == http.StatusServiceUnavailable {
 			lc.Warnf("Skipping device validation for device %s (device service %s unavailable)", device.Name, device.ServiceName)
+		} else if err.Code() == http.StatusNotFound {
+			lc.Warnf("Skipping device validation for device %s (device service %s not upgraded)", device.Name, device.ServiceName)
```

Comment:
The error message should be meaningful out of the context of this PR:
```suggestion
			lc.Warnf("Skipping device validation for device %s (device service %s <= v2.1)", device.Name, device.ServiceName)
```

Reply:
it should be `<= v2.2`

Reply:
`the device service validation API is available in v2.2 and later version`

Reply:
I suggested `<= v2.1` because several `v2.2.0-dev` versions do support it and those are smaller than v2.2. See https://semver.org/#spec-item-11

So, `<= v2.2` is not correct, both `< v2.2` and `<= v2.1` are inaccurate.

"the device service validation API is available in v2.2 and later version" sounds good to me.

```
lc.Warnf("Skipping device validation for device %s (device service %s): validation is available in v2.2 and later", 
    device.Name, device.ServiceName)
```